lib/src/common/polymer_descriptor.dart

```diff
@@ -202,7 +203,11 @@ JsObject _jsType(Type type) {
       return context['String'];
     case 'Map':
     case 'JsObject':
-    default:
       return context['Object'];
+    default:
+      var test = new JsObject(context['Object']);
```

Comment:
I was doing the wrapping so it would end up getting caught by the following.

```
    var dartClass = _getDartInstance(jsValue);
    if (dartClass != null) return dartClass;
```

Type would just end up being returned at that point.

I'm okay with any ideas you have to beef up things though.